/*
 * Generated file com/nutanix/dataprotection/v4/content.proto.
 *
 * Product version: 16.9.0-SNAPSHOT
 *
 * Part of the Nutanix Data Proetction gRPC APIs
 *
 * (c) 2024 Nutanix Inc.  All rights reserved
 *
 */

syntax = "proto3";

package com.nutanix.dataprotection.v4.content;

option java_multiple_files = true;
option java_package = "com.nutanix.dataprotection.v4.content";
option go_package = "com/nutanix/dataprotection/v4/content";

import "com/nutanix/dataprotection/v4/api_version.proto";

import "com/nutanix/dataprotection/v4/http_method_options.proto";
import "com/nutanix/dataprotection/v4/content/content.proto";

service VMRecoveryPointComputeChangedRegionsService {
    option (ntnx_api_version) = {
    MAJOR: "4",
    MINOR: "0",
    RELEASE_TYPE: "a",
    RELEASE_TYPE_REVISION: "1"
    };


  /*
   * uri: /dataprotection/v4/content/recovery-points/{recoveryPointExtId}/vm-recovery-points/{vmRecoveryPointExtId}/disk-recovery-points/{extId}/$actions/compute-changed-regions
   * http method: POST
   * Changed Regions information for the regions that have changed in the right half-open interval
   * Returns information pertaining to all the regions that have changed in the right half-open interval: [start, length).
   */
  rpc vmRecoveryPointComputeChangedRegions(VmRecoveryPointComputeChangedRegionsArg) returns (VmRecoveryPointComputeChangedRegionsRet) {
    option (ntnx_api_http) = {
      POST: "/dataprotection/v4/content/recovery-points/{recoveryPointExtId}/vm-recovery-points/{vmRecoveryPointExtId}/disk-recovery-points/{extId}/$actions/compute-changed-regions"
    };
  }
}

/*
 * message containing all attributes expected in the vmRecoveryPointComputeChangedRegions request
 */

message VmRecoveryPointComputeChangedRegionsArg {
  /*
   * The external identifier that can be used to retrieve the Recovery point using its URL.
   */
  string recovery_point_ext_id = 2001;
  /*
   * External identifier of the VM recovery point.
   */
  string vm_recovery_point_ext_id = 2002;
  /*
   * Disk Recovery point Identifier.
   */
  string ext_id = 2003;
  /*
   * VM Compute Changed Regions Request body [referenceRecoveryPointExtId, referenceVmRecoveryPointExtId,referenceDiskRecoveryPointExtId, offset, length]
   */
  com.nutanix.dataprotection.v4.content.VmRecoveryPointChangedRegionsComputeSpec body = 2004;
}

/*
 * message containing all attributes expected in the vmRecoveryPointComputeChangedRegions response
 */
message VmRecoveryPointComputeChangedRegionsRet {
  /*
   * field containing expected response content
   */
  com.nutanix.dataprotection.v4.content.ChangedRegionsModel content = 999;
  /*
   * map containing headers expected in response
   */
  map<string, string> reserved = 1000;
}